### PR TITLE
feat(console): api-score page

### PR DESCRIPTION
**Pull request by pes-spyro-soft-com**
## Issue

https://gravitee.atlassian.net/browse/APIM-5164

## Description

image

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-bingctzxoy.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/services-ngx/api-score.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TestBed } from '@angular/core/testing';
+
+import { ApiScoreService } from './api-score.service';
+
+xdescribe('ApiScoreService', () => {
```
Comment (michel-barret) [inline gravitee-apim-console-webui/src/services-ngx/api-score.service.spec.ts:21]: Shouldn't be?

```suggestion
describe('ApiScoreService', () => {
```

File: gravitee-apim-console-webui/src/management/api/api-score/api-score.component.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        <gio-card-empty-state
+          icon="rocket"
+          title="All clear"
+          [subtitle]="'There is no recommendations or issues for your API. \n Everything looks great!'"
+        ></gio-card-empty-state>
+      }
+
+      @for (apiScoreItem of apiScore.scoreLists; track apiScoreItem.name) {
+        <app-api-score-list [scoreList]="apiScoreItem"></app-api-score-list>
+      }
```
Comment (michel-barret) [inline gravitee-apim-console-webui/src/management/api/api-score/api-score.component.html:64]: Maybe that can be writen with

```
@for(...) {
    <app-api-score-list>...
} @empty {
    <gio-card-empty-state>...
}
```

File: gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe({
+        next: (apiScore) => {
+          this.apiScore = apiScore;
+        },
+      });
+  }
+
+  public evaluate() {
+    this.apiScoreService.evaluate().subscribe();
```
Comment (michel-barret) [inline gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts:66]: We don't display a message to inform that is send and results will be accessibles soon (or for the moment that not implmented but soon)?

File: gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      .get(this.activatedRoute.snapshot.params.apiId)
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe({
+        next: (api) => {
+          this.api = api;
+        },
+      });
+
+    this.apiScoreService
+      .getAllClear()
```
Comment (michel-barret) [inline gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts:56]: apiScoreService don't get the api or apiId as parameter? (I know that it's a mock but to prepare the next steps)

**Review by michel-barret — APPROVED** [review]

File: gravitee-apim-console-webui/src/services-ngx/api-score.service.spec.ts (hunk 1 above)
Comment (pes-spyro-soft-com) [reply gravitee-apim-console-webui/src/services-ngx/api-score.service.spec.ts:21]: this is empty, placeholder service, test skipped.

File: gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts (hunk 3 above)
Comment (pes-spyro-soft-com) [reply gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts:66]: to be implemented when we connect backend to it. Just placeholder for now.

File: gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts (hunk 4 above)
Comment (pes-spyro-soft-com) [reply gravitee-apim-console-webui/src/management/api/api-score/api-score.component.ts:56]: this method is a placeholder, for ID and backend implementation there will be `get` method on service

File: gravitee-apim-console-webui/src/management/api/api-score/api-score.component.html (hunk 2 above)
Comment (pes-spyro-soft-com) [reply gravitee-apim-console-webui/src/management/api/api-score/api-score.component.html:64]: good comment, however we still show api score lists (they are empty lists wiht headers) when there are no score commnets . So it is never actually empy.

**Pull request merged**